lib/watch/index.js

```diff
@@ -48,7 +48,7 @@ const watch = function (args, ignoreWatch, verboseWatch) {
     })
 
     _child.on('exit', function (code, signal) {
-      if (!code) {
+      if (code > 0) {
```

Comment:
Could you describe the intent here?

`!0` would be `true`, while `0 > 0` is `false` so I don't get why this inversion

Reply:
We can delete this code block

Reply:
It seems like the idea was that if code can either be null or an exit code integer. In the case of a > 0 code, the watcher should stop. In the case of null or 0, restart. It was ignoring > 0 codes before, mostly resulting in desired behavior. 

I can also just remove the block.

Reply:
While working through Fastify CLI generated JS and TS projects, I _did_ find after changing an app file that the `code` would be `1` in each case I tried...leading to the `stop()` call in line 52 here.  However, removing the `if` block altogether seemed to produce identical behavior.  I don't understand enough of what is going on to determine which option is correct.  I don't know if either choice might lead to dangling processes or not.

Reply:
I guess I should also note that outside the scenario of the app restarting due to a source file change, I otherwise observed that the `code` would be `0` or `null`.

Reply:
Ah ok. Sounds like we should stick with the original behavior and just remove the switch block all together then.  Sorry to add more noise, I didn't pick up on that caveat. 

If I find some time today I'll see if I can investigate the tests failing. 

I noticed that the (currently disabled) watch tests were disabled in ci due to lack of chokidar support for Linux (or windows)? I wonder, if we can restore these tests, could we run them in the macOS image?

Reply:
Cherry picked your patch that does this.